### ISSUE TITLE
Issue with AST wave generator in pull request

### ISSUE THREAD

**Issue by anardek** [UNC-CECL/brie#7]
def test_ashton_vs_WaveAngleGenerator():

    angle_array, step = np.linspace(-90.0, 90.0, 5, retstep=True)
    # angle_array, step = np.linspace(-90.0, 90.0, 181, retstep=True)

    dist = ashton(a=0.8, h=0.2, loc=-90, scale=180)
    ashton_pdf = dist.pdf(angle_array) * step
    # area_under_curve, abserr = quad(dist.pdf, -90, 90)

    waves = WaveAngleGenerator(asymmetry=0.8, high_fraction=0.2)
    waves_pdf = waves.pdf(np.deg2rad(angle_array)) * np.deg2rad(step)
    # area_under_curve, abserr = quad(waves.pdf, -np.pi / 2.0, np.pi / 2.0)

    assert waves_pdf == pytest.approx(ashton_pdf)

**Comment by anardek** [comment]
Question is: should the wave_angle pdf at -90 start at zero (WaveAngleGenerator) or wave_asymm*highness (Ashton)?

**Comment by mcflugen** [comment]
I think the *ashton* implementation is probably correct. However, in practice, it probably doesn't really matter as it would be incredibly unlikely (impossible) for a value of exactly -90 to be drawn.

I notice that `waves.pdf(90.0) > 0.0` (but `waves.pdf(-90.0) == 0.0`, as you showed), meaning the distribution is exclusive on the lower end but inclusive on the upper end. For symmetry, I would think that the distribution should either include both endpoints or exclude both endpoints, but not a mixture.

If I remember correctly, the problem you're getting at is when the pdf is sampled in 1 degree increments for the convolution. Is that correct? The code in *brie.py* that gets the wave angles is currently,
```python
wave_angles = np.linspace(-90.0, 90.0, 180)
```
This generates angles for -90 to 90 (inclusive) with a spacing of *181 / 180* (so, just a little larger than a degree). One easy fix would be to change this to something like,
```python
wave_angles = np.linspace(-89.5, 89.5, 180)
```
The wave angles are now in the middle of the bins, symmetrical about zero, spaced by 1 degree, and, since the endpoints aren't included in the *wave_angles* array, both distributions should match.

**Comment by anardek** [comment]
Thanks for the wave_angles explanation: understood. 

Since the bounds in our waves.pdf are not symmetric, should we retrofit waves.pdf to match _ashton_? What was your vision here? Just include the ashton wave generator for users, or incorporate it somehow in the code? I think it is just currently used for testing, right?

**Comment by anardek** [comment]
As we discussed last week, I decided to move forward with the WaveAngleGenerator and a wave angle array spaced by 1 degree between -89.5 and 89.5. I'm going to commit my code as is, but unfortunately Jaap used a lot of indexing (for example in coast_diff) that does not work for bounds other than [-90,90], so all of the coast_diff tests are failing. After I commit, I modify the code and tests to accept the ashton distribution instead.
